### ISSUE TITLE
💅 noComponentHookFactories: use*-named non-function bindings are flagged as hooks

### ISSUE THREAD

**Issue by maks-oleksyuk** [biomejs/biome#11335]
`lint/nursery/noComponentHookFactories` reports any variable named with the hook naming convention (`use` + uppercase) as "a hook defined inside a function", even when the value is not a function at all — a plain boolean, string, or any other non-callable value.

Root cause: for the hook path specifically, the rule detects hooks purely by name and reports immediately, with no check that the binding is actually a function. This is different from the component path, which does call the heavier `ReactComponentInfo::from_declaration` check before reporting.

https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/lint/nursery/no_component_hook_factories.rs#L112-L120

`is_react_hook_name` is a pure string check:

https://github.com/biomejs/biome/blob/main/crates/biome_js_analyze/src/react/hooks.rs#L119-L121

## Environment information

<details>

```bash
CLI:
  Version:                      2.5.8
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:                npm
  JS_RUNTIME_VERSION:           v26.7.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/11.19.0
```

</details>

## Rule name

noComponentHookFactories

## Playground link

https://biomejs.dev/playground/?lintRules=noComponentHookFactories&tab=formatter&pane=Diagnostics&code=ZgB1AG4AYwB0AGkAbwBuACAAZgBhAGMAdABvAHIAeQAoACkAIAB7AAoAIAAgAGMAbwBuAHMAdAAgAHUAcwBlAEMAbwBsAG8AcgBzACAAPQAgAHQAcgB1AGUAOwAKACAAIAByAGUAdAB1AHIAbgAgAHUAcwBlAEMAbwBsAG8AcgBzADsACgB9AAoA&language=ts

Verified live: `const useColors = true;` inside a plain function triggers `Do not define hook useColors inside a function.` even though there is no function, hook, or JSX anywhere in the file.

## Expected result

A `use`-prefixed binding should only be reported as a hook when it's actually a function (arrow function, function expression, or function declaration) — a boolean/string/number/etc. assigned to a `use*`-named variable is not a hook and shouldn't be flagged.

- [x] I agree to follow Biome's Code of Conduct
